buildSrc/src/main/kotlin/io/spine/internal/gradle/github/pages/GitHubPagesBranch.kt

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.github.pages
+
+import io.spine.internal.gradle.Cli
+import io.spine.internal.gradle.RepoSlug
+import java.io.File
+import org.gradle.api.GradleException
+
+/**
+ * Represents the branch used for publishing documentation.
+ *
+ * The repository's GitHub URL is derived from the `REPO_SLUG` environment variable.
+ * The branch dedicated to publishing documentation is automatically checked out in
+ * this repository. Also, the username and email of the git committer are
+ * automatically configured. The username is set to "UpdateGitHubPages Plugin", and
+ * the email is derived from the `FORMAL_GIT_HUB_PAGES_AUTHOR` environment variable.
+ *
+ * This class provides functionality to commit changes in the specific directory and
+ * push them to the remote repository.
+ */
+internal class GitHubPagesBranch {
+
+    /** The branch dedicated to publishing documentation. */
+    private val targetBranch = "gh-pages"
+    val repoFolder: File
+
+    internal constructor(rootFolder: File, repoFolder: File) {
+        this.repoFolder = repoFolder
+
+        SshKey(rootFolder).register()
+        clone(rootFolder)
+        checkout()
+        configureCommitter()
+    }
+
+    /**
+     * Clones the Git repository with the GitHub URL retrieved from the `REPO_SLUG`
+     * environment variable.
+     */
+    private fun clone(rootFolder: File) {
```

Comment:
I don't understand this either. Perhaps, an unfortunate naming choice for `rootFolder`. Why "root"?

Also, the docs says nothing about the destination of the clone procedure. I bet you would want to know this tiny detail as a docs reader.